hearth/sig/lib/hearth/telemetry/base.rbs

```diff
@@ -0,0 +1,46 @@
+module Hearth
+  module Telemetry
+    class TelemetryProviderBase
+      def initialize: (?tracer_provider: TracerProviderBase?, ?context_manager: ContextManagerBase) -> void
+      attr_reader tracer_provider: TracerProviderBase?
+
+      attr_reader context_manager: ContextManagerBase
+    end
+
+    class TracerProviderBase
+      def tracer: (?String name) -> TracerBase
+    end
+
+    class TracerBase
+      def start_span: (String name, ?untyped with_parent, ?Hash[String, untyped] attributes, ?untyped kind) -> SpanBase
+
+      def in_span: (String name, ?Hash[String, untyped] attributes, ?untyped kind) -> SpanBase
```

Comment:
Are all of these types filled out? I would think there's a type to kind, status, exception, etc